legend-sdlc-server/src/main/java/org/finos/legend/sdlc/server/gitlab/api/GitLabProjectConfigurationApi.java

```diff
@@ -401,6 +405,40 @@ public List<MetamodelDependency> getMetamodelDependencies()
         return ProjectStructure.getProjectStructure(projectConfiguration).getAvailableGenerationConfigurations();
     }
 
+    @Override
+    public ProjectConfigurationStatusReport checkProjectConfigurationStatus(String projectId)
+    {
+        return new ProjectConfigurationStatusReport()
+        {
+            @Override
+            public boolean isProjectConfigured()
+            {
+                return getProjectConfiguration(projectId) != null;
+            }
+
+            @Override
+            public String getReviewId()
+            {
+                try
+                {
+                    List<MergeRequest> mergeRequests = getGitLabApi().getMergeRequestApi().getMergeRequests(new MergeRequestFilter());
+                    GitLabProjectId gitLabProjectId = parseProjectId(projectId);
+                    List<MergeRequest> projectConfigurationMergeRequest = mergeRequests.stream().filter(mergeRequest -> mergeRequest.getSourceProjectId().equals(gitLabProjectId.getGitLabId()) && mergeRequest.getSourceBranch().contains(GitLabProjectApi.PROJECT_CONFIGURATION_WORKSPACE_ID_PREFIX)).collect(Collectors.toList());
+                    if (projectConfigurationMergeRequest.size() > 0)
+                    {
+                        return projectConfigurationMergeRequest.get(0).getIid().toString();
+                    }
+                    return null;
+                }
+                catch (GitLabApiException e)
+                {
+                    throw buildException(e,
+                            () -> "User " + getCurrentUser() + " is not allowed to access the project configuration status of " + projectId);
```

Comment:
This message is only appropriate for a 403 status. You should have different messages for different cases.